coffee-chats/src/main/java/com/google/step/coffee/data/CalendarUtils.java

```diff
@@ -0,0 +1,82 @@
+package com.google.step.coffee.data;
+
+import static com.google.step.coffee.APIUtils.APPLICATION_NAME;
+import static com.google.step.coffee.APIUtils.HTTP_TRANSPORT;
+import static com.google.step.coffee.APIUtils.JSON_FACTORY;
+
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.ConferenceData;
+import com.google.api.services.calendar.model.ConferenceSolutionKey;
+import com.google.api.services.calendar.model.CreateConferenceRequest;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.EventAttendee;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.step.coffee.OAuthService;
+import com.google.step.coffee.entity.TimeSlot;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/** Utilities to access Google Calendar API for an authorised user. */
+public class CalendarUtils {
+  private static final UserStore userStore = new UserStore();
+
+  public static final List<String> SCOPES = Arrays.asList(CalendarScopes.CALENDAR_SETTINGS_READONLY,
+      CalendarScopes.CALENDAR_READONLY, CalendarScopes.CALENDAR_EVENTS);
+
+  /** Add given event to user's primary calendar. */
+  public static void addEvent(String userId, Event event) {
+    Calendar service = getCalendarService(userId);
+
+    try {
+      service.events().insert("primary", event).setConferenceDataVersion(1).execute();
+    } catch (IOException e) {
+      System.out.println("Event could not be created: " + e.getMessage());
```

Comment:
Ignoring exceptions is suspicious, but if this is a temporary solution then I guess it's fine.

Reply:
This exception would be raised if the calendar event could not be created by the API. I am not sure how else to handle this appropriately, apart from triggering a retry if the exception was caught in the calling function. I have added functionality for the request to be attempted a second time on failure before returning.

Reply:
Any tests for the retry?